packages/cactus-plugin-ccmodel-hephaestus/src/main/typescript/plugin-ccmodel-hephaestus.ts

```diff
@@ -0,0 +1,844 @@
+import {
+  BesuV2TxReceipt,
+  EthereumTxReceipt,
+  FabricV2TxReceipt,
+} from "./models/transaction-receipt";
+import { millisecondsLatency } from "./models/utils";
+import { RunTransactionV1Exchange as RunTransactionV1ExchangeBesu } from "@hyperledger/cactus-plugin-ledger-connector-besu";
+import { RunTransactionV1Exchange as RunTransactionV1ExchangeEth } from "@hyperledger/cactus-plugin-ledger-connector-ethereum";
+import { RunTxReqWithTxId } from "@hyperledger/cactus-plugin-ledger-connector-fabric";
+
+import { Observable } from "rxjs";
+import { filter, tap } from "rxjs/operators";
+
+export interface IPluginCcModelHephaestusOptions extends ICactusPluginOptions {
+  connectorRegistry?: PluginRegistry;
+  logLevel?: LogLevelDesc;
+  webAppOptions?: IWebAppOptions;
+  instanceId: string;
+  ethTxObservable?: Observable<RunTransactionV1ExchangeEth>;
+  besuTxObservable?: Observable<RunTransactionV1ExchangeBesu>;
+  fabricTxObservable?: Observable<RunTxReqWithTxId>;
+  sourceLedger: LedgerType;
+  targetLedger: LedgerType;
+  ccLogsDir?: string;
+}
+
+export class CcModelHephaestus implements ICactusPlugin, IPluginWebService {
+  private readonly log: Logger;
+  private readonly instanceId: string;
+  private endpoints: IWebServiceEndpoint[] | undefined;
+  private crossChainLog: CrossChainEventLog;
+  private unmodeledEventLog: CrossChainEventLog;
+  private nonConformedCrossChainLog: CrossChainEventLog;
+  private readonly nonConformedCCTxs: string[];
+  private crossChainModel: CrossChainModel;
+  public readonly className = "plugin-ccmodel-hephaestus";
+  private caseID: string;
+  private readonly besuTxObservable?: Observable<RunTransactionV1ExchangeBesu>;
+  private readonly ethTxObservable?: Observable<RunTransactionV1ExchangeEth>;
+  private readonly fabricTxObservable?: Observable<RunTxReqWithTxId>;
+  private readonly sourceLedger: LedgerType;
+  private readonly targetLedger: LedgerType;
+  private startMonitoring: number | null = null;
+  private isModeling: boolean;
+  private readonly ccLogsDir: string;
+
+  constructor(public readonly options: IPluginCcModelHephaestusOptions) {
+    const startTime = new Date();
+    const fnTag = `PluginCcModelHephaestus#constructor()`;
+    if (!options) {
+      throw new Error(`${fnTag} options falsy.`);
+    }
+    Checks.truthy(options.instanceId, `${fnTag} options.instanceId`);
+    const level = this.options.logLevel || "INFO";
+    const label = this.className;
+    this.log = LoggerProvider.getOrCreate({
+      label: label,
+      level: level,
+    });
+    this.instanceId = this.options.instanceId;
+    this.crossChainLog = new CrossChainEventLog({
+      name: "HEPHAESTUS_EVENT_LOGS",
+    });
+    this.unmodeledEventLog = new CrossChainEventLog({
+      name: "HEPHAESTUS_UNMODELED_LOGS",
+    });
+    this.nonConformedCrossChainLog = new CrossChainEventLog({
+      name: "HEPHAESTUS_NON_CONFORMANCE_LOGS",
+    });
+
+    this.caseID = "UNDEFINED_CASE_ID";
```

Comment:
Is it safe to use no-unique string as a default caseID? Wouldn't it collide if we don't set the caseID ourselves? Maybe we could set this as some uuid by default?

Reply:
Agreed